website/docs/d/stream_connection.html.markdown

```diff
@@ -0,0 +1,57 @@
+sidebar_current: "docs-mongodbatlas-datasource-stream-connection"
+description: |-
+    Describes a Stream Connection.
+---
+
+# Data Source: mongodbatlas_stream_connection
+
+`mongodbatlas_stream_connection` describes a stream connection.
+
+-> **NOTE:** Atlas Streams is currently in Preview, you'll need to set the environment variable `MONGODB_ATLAS_ENABLE_BETA=true` to use this data source. To learn more about stream processing and existing limitations, see the [Atlas Stream Processing Documentation](https://www.mongodb.com/docs/atlas/atlas-sp/overview/#atlas-stream-processing-overview).
+
+## Example Usage
+
+```terraform
+data "mongodbatlas_stream_connection" "example" {
+    project_id = "<PROJECT_ID>"
+    instance_name = "<INSTANCE_NAME>"
+    connection_name = "<CONNECTION_NAME>"
+}
+```
+
+## Argument Reference
+
+* `project_id` - (Required) Unique 24-hexadecimal digit string that identifies your project.
+* `instance_name` - (Required) Human-readable label that identifies the stream instance.
+* `connection_name` - (Required) Human-readable label that identifies the stream connection.
+
+## Attributes Reference
+
+* `type` - Type of connection. Can be either `Cluster` or `Kafka`.
+
+If `type` is of value `Cluster` the following additional attributes are defined:
+* `cluster_name` - Name of the cluster configured for this connection.
+
+If `type` is of value `Kafka` the following additional attributes are defined:
+* `authentication` - User credentials required to connect to a Kafka Cluster. Includes the authentication type, as well as the parameters for that authentication mode. See [authentication](#authentication).
+* `bootstrap_servers` - Comma separated list of server addresses.
+* `config` - A map of Kafka key-value pairs for optional configuration. This is a flat object, and keys can have '.' characters.
+* `security` - Properties for the secure transport connection to Kafka. For SSL, this can include the trusted certificate to use. See [security](#security).
+
+### Authentication
+
+* `mechanism` - Style of authentication. Can be one of `PLAIN`, `SCRAM-256`, or `SCRAM-512`.
+* `username` - Username of the account to connect to the Kafka cluster.
+* `password` - Password of the account to connect to the Kafka cluster.
+
+### Security
+
+* `broker_public_certificate` - A trusted, public x509 certificate for connecting to Kafka over SSL. String value of the certificate must be defined in the attribute.
+* `protocol` - Describes the transport type. Can be either `PLAINTEXT` or `SSL`.
+
+
+To learn more, see: [MongoDB Atlas API - Stream Connection](https://www.mongodb.com/docs/atlas/reference/api-resources-spec/#tag/Streams/operation/getStreamConnection) Documentation.
+The [Terraform Provider Examples Section](https://github.com/mongodb/terraform-provider-mongodbatlas/blob/master/examples/atlas-streams/README.md) also contains details on the overall support for Atlas Streams in Terraform.
```

Comment:
referenced README file is defined in https://github.com/mongodb/terraform-provider-mongodbatlas/pull/1752